src/libs/auth.js

```diff
@@ -155,6 +151,14 @@ authStore.subscribe(withErrorReporting('Error checking TOS', async (state, oldSt
   }
 }))
 
+authStore.subscribe(async (state, oldState) => {
+  if (!oldState.acceptedTos && state.acceptedTos) {
+    window.Appcues && window.Appcues.identify(state.user.id, {
+      dateJoined: parseJSON((await Ajax().User.firstTimestamp()).timestamp).getTime()
```

Comment:
Hmm... I think there's a problem here. `User.firstTimestamp()` doesn't just return the data. It also is (almost) the only path to writing the data in the first place! I didn't notice that when this was previously made conditional on `window.Appcues`.

It looks like `NpsSurvey` also calls this on mount when a user is registered. Even then, it's buried in the conditional for determining whether or not to show the NPS prompt.

There may not be any problem with the changes in this PR, but I'm now even more uncomfortable with how much effort it took to analyze this and how coincidental all of this behavior feels.

Reply:
The `firstTimestamp` bit is weird to begin with, since it's not something that the UI should be responsible for in the first place. The logic there is such that we'll always call it at least once per user, and that was the original usage of this data anyhow. Again, it's written in a very specific way, but given the use case I don't see a way to simplify it much.